s3/pkg/service/multipart.go

```diff
@@ -306,6 +327,40 @@ func (s *s3Service) CompleteMultipartUpload(ctx context.Context, in *pb.Complete
 		return err
 	}
 
+	md5Writer := md5.New()
+	var totalSize int64 = 0
+	for i := 0; i < len(in.CompleteParts); i++ {
+		if in.CompleteParts[i].PartNumber != int64(i+1) {
+			log.Infof("wrong order for part number. ")
+			err = ErrInvalidPart
+			return err
+		}
+		part, ok := multipart.Parts[i+1]
+		if !ok {
+			log.Infoln("missed object part. partno:", i)
+			err = ErrInvalidPart
+			return err
+		}
+
+		if part.Etag != in.CompleteParts[i].ETag {
```

Comment:
Is it better to check the ETag in each UploadPart API instead of in CompleteMultipart API?

Reply:
CompleteParts[i].ETag is from client, so it is necessary to check if the ETag from user is correct.

Reply:
OK